PAM_app/app/src/main/java/com/example/pam_app/cosmetics/CosmeticsActivity.kt

```diff
@@ -16,68 +16,64 @@ import com.example.pam_app.cosmetics.presenter.Presenter
+    lateinit var recyclerView: RecyclerView
+    private val adapter by lazy { MainAdapter() }
+    private lateinit var presenter: Presenter
 
-    class CosmeticsActivity : AppCompatActivity(), View {
-
-        private lateinit var binding: ActivityCosmeticsBinding
-        lateinit var recyclerView: RecyclerView
-        private val adapter by lazy { MainAdapter() }
-        private lateinit var presenter: Presenter
-
-        override fun onCreate(savedInstanceState: Bundle?) {
-            super.onCreate(savedInstanceState)
-            binding = ActivityCosmeticsBinding.inflate(layoutInflater)
-            val view = binding.root
-            setContentView(view)
-
-            presenter = Presenter(this, Model(Repository()))
-            setupRecyclerview()
-            presenter.initView()
-        }
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityCosmeticsBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+        presenter = Presenter(this, Model(Repository()))
+        setupRecyclerview()
+        presenter.initView()
+    }
 
-        override fun init() {
-            System.out.println("beggin post request")
-            val myPost = Cosmetics(
-                "https://cdn.shopify.com/s/files/1/1338/0845/products/brain-freeze_a_800x1200.jpg?v=1502255076",
-                "Lipstick", 10.2F
-            )
-            presenter.putItem(myPost)
+    override fun init() {
+        System.out.println("beggin post request")
+        val myPost = Cosmetics(
+            "https://cdn.shopify.com/s/files/1/1338/0845/products/brain-freeze_a_800x1200.jpg?v=1502255076",
+            "Lipstick", 10.2F
+        )
+        presenter.putItem(myPost)
 
-            val myPost2 = Cosmetics(
-                "https://www.purpicks.com/wp-content/uploads/2018/02/Ombre-Amazonie-CC.png",
-                "Eyeshadow", 5.2F
-            )
-            presenter.postSmth(myPost2)
-            presenter.getBlush()
-            presenter.getLipstick()
-        }
+        val myPost2 = Cosmetics(
+            "https://www.purpicks.com/wp-content/uploads/2018/02/Ombre-Amazonie-CC.png",
+            "Eyeshadow", 5.2F
+        )
+        presenter.postSmth(myPost2)
+        presenter.getBlush()
+        presenter.getLipstick()
+    }
 
-        override fun showGetResponse(response: Response<List<Cosmetics>>) {
-            if (response.isSuccessful) {
-                response.body()?.let { adapter.setData(it) }
-            } else {
-                Toast.makeText(this, response.code().toString(), Toast.LENGTH_SHORT).show()
-            }
+    override fun showGetResponse(response: Response<List<Cosmetics>>) {
+        if (response.isSuccessful) {
+            response.body()?.let { adapter.setData(it) }
+        } else {
+            Toast.makeText(this, response.code().toString(), Toast.LENGTH_SHORT).show()
```

Comment:
This is actually presentation business logic. It should be moved to presenter, in that way it could be covered with unit tests.